packages/api/endpoints/providers.js

```diff
@@ -129,22 +178,26 @@ async function put({ params: { id }, body }, res) {
  */
 async function del(req, res) {
   const providerModel = new Provider();
+  const knex = await getKnexClient({ env: process.env });
 
   try {
-    await providerModel.delete({ id: req.params.id });
-
-    if (inTestMode()) {
-      const esClient = await Search.es(process.env.ES_HOST);
-      await esClient.delete({
-        id: req.params.id,
-        type: 'provider',
-        index: process.env.ES_INDEX,
-      }, { ignore: [404] });
-    }
+    await knex.transaction(async (trx) => {
+      await trx(tableNames.providers).where({ name: req.params.id }).del();
```

Comment:
Don't directly translate here

Reply:
Contemplating this further, running a translation on the index directly seems 'ok' here, over faking a translation to index name->id.